.circleci/config.yml

```
@@ -253,6 +253,89 @@ jobs:
                 include_job_number_field: false
                 include_project_field: false
                 failure_message: ':red_circle: Scheduled tests failed on iOS device!'
+  build-android-js-bundle:
+    # This job builds the JS bundle that the Android react-navite-bridge
+    # bridges and saves it in the CI cache to make it available to other jobs,
+    # in particular the one building the AAR for binary distribution.
+    #
+    # Running on a Ubuntu machine to have more memory compared to the Docker +
+    # Node image that CircleCI offers (circleci/node)
+    #
+    # See: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3038#issuecomment-765334675
+    machine:
+      image: << pipeline.parameters.linux-machine-image >>
+    steps:
+      - checkout
+      - checkout-submodules
+      - npm-install
+      - run:
+          name: Build JS Bundle
+          command: |
+            npm run bundle:android
+          # This command may take a long time, so prevent CircleCI from killing
+          # the build if it doesn't receive output for more than its default
+          # ten minutes.
+          no_output_timeout: 30m
+      - run:
+          name: Get cache id
+          command: |
+            echo $CIRCLE_SHA1 > cache_id
+            cat cache_id
+      - save_cache:
+          name: Cache React Native Android Bridge JS Bundle
+          key: android-js-bundle-{{ checksum "cache_id" }}
+          paths:
+            # The build process for the react-navite-bridge requires the App.js
+            # bundle and also access to the node_modules folder.
```

Should I a link to the line of code that requires `node_modules` diff on GitHub once the PR for that is merged?

---

```suggestion
            # The build process for the react-native-bridge requires the App.js and the node_modules folder.
```

We technically don't need the `node_modules` once the bundle is created. However, without it [configuration gets a bit messy](https://github.com/WordPress/gutenberg/pull/29488#discussion_r590635389). I don't think it's worth adding a link here and I am hoping we can work towards not needing the `node_modules` folder.